lib/path.dart

```diff
@@ -324,6 +340,19 @@ String relative(String path, {String from}) =>
 ///     path.isWithin('/root/path', '/root/path') // -> false
 bool isWithin(String parent, String child) => context.isWithin(parent, child);
 
+/// Returns `true` if [path1] points to the same location as [path2], and
+/// `false` otherwise.
+///
+/// The [hash] function returns a hash code that matches these equality
+/// semantics.
+bool equals(String path1, String path2) => context.equals(path1, path2);
+
+/// Returns a hash code for [path] that matches the semantics of [equals].
```

Comment:
How about adding "such that two paths that are equivalent will always have the same hash code."